app/components/documents/document-card.hbs

```diff
@@ -138,6 +139,7 @@
                 @disabled={{this.loadPieceRelatedData.isRunning}}
                 @alignment="right"
               >
+              {{#if this.mayEditOrDelete}}
                 <AuButton
```

Comment:
hiding the option is not needed as the dropdown itself is hidden
There are 4 actions in this dropdown (3 edit, 1 link to signFlow preview)
when signFlow exists, 3 options are hidden by the boolean and the 4th is mutually exclusive so this would result in a dropdown with no options (which throws errors at the moment)